### PR TITLE
Fix failing unit test BackupModifyRestore.

### PR DESCRIPTION
**Pull request by danielfenner**
The xsave command is only saving state components that have been
initialized. Registers in uninitilized state are not backed up.

Looks like the fpu has previoulsly been used by something linked into
the tests and now this does not happen anymore. The fix explicitly
touches the fpu registers such that they get backed up.

### CODE REVIEW

**Review by antonrohr — APPROVED** [review]
lgtm!

File: src/UserSpaceInstrumentation/RegisterStateTest.cpp [hunk 1]
```diff
@@ -30,11 +30,12 @@ void Child() {
   for (size_t i = 0; i < avx_bytes.size(); ++i) {
     avx_bytes[i] = i;
   }
-  // The first two lines move the memory to the registers. The "%0" and "%1" refer to the addresses
-  // of "rax" and "avx_bytes" given in the second line from the bottom. "int3" just is the
-  // breakpoint. The parent does "waitpid" for that. Line four and five move the registers back into
-  // memory so they are available for verification below.
+  // The first three touch the fpu and move the memory to the registers. The "%0" and "%1" refer to
```
Comment (dpallotti) [inline src/UserSpaceInstrumentation/RegisterStateTest.cpp:33]: three -> three lines/instructions ?

File: src/UserSpaceInstrumentation/RegisterStateTest.cpp [hunk 2]
```diff
@@ -30,11 +30,12 @@ void Child() {
...
+  // the addresses of "rax" and "avx_bytes" given in the second line from the bottom. "int3" just is
+  // the breakpoint. The parent does "waitpid" for that. Line four and five move the registers back
```
Comment (dpallotti) [inline src/UserSpaceInstrumentation/RegisterStateTest.cpp:35]: Line four and five -> Line five and six

**Review by dpallotti — APPROVED** [review]
Looks good otherwise.

**Pull request merged**